src/tracks/ui/TrackView.h

```diff
@@ -88,8 +91,11 @@ class AUDACITY_DLL_API TrackView /* not final */ : public CommonTrackCell
    // Private factory to make appropriate object; class TrackView handles
    // memory management thereafter
    virtual std::shared_ptr<TrackVRulerControls> DoGetVRulerControls() = 0;
+   // May return nullptr if track does not need affordance area
+   virtual std::shared_ptr<CommonTrackCell> DoGetAffordanceControls();
```

Comment:
This is not a pure virtual:  you should commend that the default behavior is to return null.